services/subnet_manager/src/test/java/com/futurewei/alcor/subnet/SubnetControllerTests.java

```diff
@@ -138,13 +148,50 @@ public void createSubnetState_canNotFindRoute_notPass () throws Exception {
                 UnitTestConfig.vpcId, UnitTestConfig.name, UnitTestConfig.cidr, new ArrayList<RouteWebObject>(){{add(new RouteWebObject());}});
 
         VpcStateJson vpcStateJson = new VpcStateJson(vpcState);
+        MacStateJson macResponse = new MacStateJson();
 
         Mockito.when(subnetDatabaseService.getBySubnetId(UnitTestConfig.subnetId))
                 .thenReturn(subnetState);
         Mockito.when(subnetService.verifyVpcId(UnitTestConfig.projectId, UnitTestConfig.vpcId))
                 .thenReturn(vpcStateJson);
-        Mockito.when(subnetService.createRouteRules(UnitTestConfig.subnetId, subnetState))
-                .thenReturn(null);
+        Mockito.when(subnetService.createRouteRules(eq(UnitTestConfig.subnetId), any(SubnetState.class)))
+                .thenThrow(new FallbackException("fallback request"));
+        Mockito.when(subnetService.allocateMacGateway(anyString(), anyString(), anyString()))
+                .thenReturn(macResponse);
+        try {
+            this.mockMvc.perform(post(creatwUri).contentType(MediaType.APPLICATION_JSON)
+                    .content(UnitTestConfig.resource))
+                    .andDo(print())
+                    .andExpect(status().is(201))
+                    .andExpect(MockMvcResultMatchers.jsonPath("$.subnet.id").value(UnitTestConfig.subnetId));
+        }catch (Exception ex) {
+            //System.out.println(ex.getMessage());
+            assertEquals(UnitTestConfig.createException, ex.getMessage());
+        }
+
+    }
+
+    @Test
+    public void createSubnetState_canNotFindMac_notPass () throws Exception {
```

Comment:
Looks like we have some fall back UTs. Good job! 

Recommend to cover all possible fall back failures.

Reply:
NP